api/docs/mapstruct.md

```diff
@@ -0,0 +1,608 @@
+        employeeModel.setSalary( toAnnalSalary( employeeDbEntity.getHourlyRate() ) );
+        employeeModel.setAddress( employeeDbEntity.getAddress() );
+
+        return employeeModel;
+    }
+}
+```
+
+### Testing
+It's pretty easy to add a test class for the mapper. Here's a basic test case for the happy path.
+If your input class has any nullable parameters or fields on those parameters, or there are any other
+edge cases, you'd want to test those as well.
+```java
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.utils.mappers.examples.Department;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeDbEntity;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeMapper;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeMapperImpl;
+import org.pmiops.workbench.utils.mappers.examples.EmployeeModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class EmployeeMapperTest {
+
+  public static final String ADDRESS = "123 Sesame Street";
+  @Autowired private EmployeeMapper employeeMapper;
+
+  @TestConfiguration
+  @Import({EmployeeMapperImpl.class})
+  static class Configuration {}
+
+  @Test
+  public void testToModel() {
+    final EmployeeDbEntity employeeDbEntity = new EmployeeDbEntity("John", "Doe", 2, 3.00, ADDRESS);
+
+    final EmployeeModel employeeModel = employeeMapper.toModel(employeeDbEntity);
+    assertThat(employeeModel.getDepartment()).isEqualTo(Department.IT);
+    assertThat(employeeModel.getSalary()).isWithin(1.0e-6).of(6000.0);
+    assertThat(employeeModel.getName()).isEqualTo("John Doe");
+    assertThat(employeeModel.getAddress()).isEqualTo(ADDRESS);
+  }
+}
+```
+## Tips & Strategies
+
+### Stub it so it builds, and go one property at a time
+For big target classes, put in some temporary `ignore` statements or other hacks just to get it
+to the point that it generates the implemmentation class and actually compiles without errors. Then
+proceed one property at a time until all the warnings are gone.
+ 
+### Proofread the generated code
+First, you should *always* look at the generated code and step through it in the debugger. This is
+because, while the codegen should be deterministic, most of us don't go around with its order of
+operator precedence in our head. Particularly if you have many input parameters with multiple fields
+with common names, you may find that `foo.id` is getting used instead of `bar.id` becase one's a 
+`long` and the other an `int`. Or something. 
+
+In general, the library gives very reasonable error
+messages and is very cranky at compile time. This is A Good Thing.
+
+## Write tests for mappers
```

Comment:
Future idea: It would be nice if these Java warning for missed properties could be asserted in a simple resuable test

Reply:
Even better, we can set up a [`@MapperConfig`](https://mapstruct.org/documentation/stable/api/org/mapstruct/MapperConfig.html) class and set the [unmappedTargetPolicy](https://mapstruct.org/documentation/stable/api/org/mapstruct/MapperConfig.html#unmappedTargetPolicy--) on it globally to [ERROR](https://mapstruct.org/documentation/1.0/api/org/mapstruct/ReportingPolicy.html) (as well as specify Spring and a couple other things that are tedious to set everywhere, like the null source policy.

Filed https://precisionmedicineinitiative.atlassian.net/browse/RW-4926